community/kernel/src/test/java/org/neo4j/test/Race.java

```diff
@@ -118,5 +121,14 @@ public void run()
                 throw e;
             }
         }
+
+        private void randomlyDelaySlightly()
+        {
+            int target = ThreadLocalRandom.current().nextInt( 1_000_000_000 );
+            for ( int i = 0; i < target; i++ )
+            {
+                i = i;
+            }
+        }
```

Comment:
Computing random numbers and storing them in a volatile field, would be a better approach, because this loop should not be hard for the JIT to eliminate.